### PR TITLE
Feat/editable-label

### PR DESCRIPTION
**Pull request by dontry**
This MR enables user to edit message labels.
- Create new component `<MessageLabel />` and use it in  `Message` and `SelftInvocation` component
- Create a reusable hook `useEditLabel` containing event handlers for editing labels.
- Refactor `<ConditionLabel />` to use `useEditLabel` hook
- Add a new regex `removeSpacesBeforeAndInsideBrackets` for `getFormattedText` function
- Add test cases to cypress e2e test
- Update github actions e2e.yml to fix Python dependencies missing issue

TODO: fix overflow issue of the first method label upon first time edit

### CODE REVIEW

**Comment by MrCoder** [comment]
Layout is not updated:

image

image

**Review by MrCoder — CHANGES_REQUESTED** [review]
Layout is not updated. See images in comments.

File: src/components/DiagramFrame/SeqDiagram/MessageLayer/MessageLabel.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <label
+    title="Double click to edit"
+    class="px-1 cursor-text right hover:text-skin-message-hover hover:bg-skin-message-hover"
+    :class="{
+      'absolute right-1/2 translate-x-1/2 bottom-0  py-1 px-2 ml-1 cursor-text':
```
Comment (dontry) [inline src/components/DiagramFrame/SeqDiagram/MessageLayer/MessageLabel.vue:6]: `absolute right-1/2 translate-x-1/2`  rules ensure that the editing text input is centered and does not push the message arrow out of position.

File: src/components/DiagramFrame/SeqDiagram/MessageLayer/Block/Statement/Interaction/Occurrence/Occurrence.vue [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default {
       > .message
   ) {
   bottom: -17px; /* Move the absolutely positioned return message to the bottom. -17 to offset the padding of Occurrence. */
-  height: 0;
+  /* height: 0; */
```
Comment (dontry) [inline src/components/DiagramFrame/SeqDiagram/MessageLayer/Block/Statement/Interaction/Occurrence/Occurrence.vue:105]: This rule is removed as it makes the style-panel overlaps the editable return message due to the message container height being zero.

**Comment by MrCoder** [comment]
It does not wrap the method with double quotes if there are spaces or special characters.

image

**Comment by MrCoder** [comment]
修改message之后，没有刷新Layout
image

**Comment by dontry** [comment]
> 修改message之后，没有刷新Layout image

没有办法复现问题。稍微做了一些优化改动。

**Comment by MrCoder** [comment]
只有在非常特定的DSL下才可以重现。

```
title Order Service (Demonstration only)
@Actor Client
@Boundary OrderController
@EC2 <<BFF>> OrderService
group BusinessService {
  @Lambda PurchaseService
  @AzureFunction InvoiceService
}

@Starter(Client)
//`POST /orders`
OrderController.post(payload) {
  OrderService.create(payload) {
    order = new Order(payload)
    if(order != null) {
      par {
        PurchaseService.createPO(order)
        InvoiceService.createInvoice(order)      
      }      
    }
  }
}
```

https://github.com/mermaid-js/zenuml-core/assets/169417/d0cf2098-ac08-4011-965a-eaeb5617464b

**Comment by dontry** [comment]
> 修改message之后，没有刷新Layout image

这个问题是因为我掉到一个坑。之前定义的regex有一个g代表全局搜索，是有状态的。造成第一次碰到特殊字符能识别，之后搜索是从上一次得到的index算起，导致后面的检测不成功。现在这个bug已经修复了。其他你碰到的问题，我没发复现。你试试再拉一下最新的commit测试一下？有问题的情况，可以附上DSL, 我可以做regression test

**Comment by dontry** [comment]
我决定添加一些e2e测试到smoke test

**Review by MrCoder — CHANGES_REQUESTED** [review]
Similar issues on Self message. See the last comment.

**Comment by dontry** [comment]
> Similar issues on Self message. See the last comment.

Fixed

File: src/core.ts [hunk 3]
```diff
@@ -104,7 +104,6 @@ export default class ZenUml implements IZenUml {
   async doRender(config: Config | undefined) {
     console.debug("rendering start");
     const start = getStartTime();
-    clearCache();
```
Comment (MrCoder) [inline src/core.ts:107]: @danshuitaihejie 这个地方删掉有没有别的影响？

Comment (danshuitaihejie) [reply src/core.ts:107]: > @danshuitaihejie 这个地方删掉有没有别的影响？

这个地方不能的clearCache不能删除

**Review by MrCoder — APPROVED** [review]

File: src/components/DiagramFrame/SeqDiagram/MessageLayer/Block/Statement/Interaction/SelfInvocation/SelfInvocation.vue [hunk 4]
```diff
@@ -7,7 +7,7 @@
     ref="messageRef"
   >
     <label
-      class="name group px-px hover:text-skin-message-hover hover:bg-skin-message-hover relative min-h-[1em]"
+      class="name group px-px hover:text-skin-message-hover hover:bg-skin-message-hover relative min-h-[1em] w-full"
```
Comment (MrCoder) [inline src/components/DiagramFrame/SeqDiagram/MessageLayer/Block/Statement/Interaction/SelfInvocation/SelfInvocation.vue:10]: This is causing an undesired effect see below video. This does not seem to be the cause if the issue. There is real issue that needs to be fixed. The corresponding interaction should not have width at the first place.

https://github.com/mermaid-js/zenuml-core/assets/169417/1b866957-c33d-4fd0-aaf8-0cbd3e21d0c2

**Pull request merged**